### ISSUE TITLE
Investigate a solution for continuous deployment

### ISSUE THREAD

**Issue by RichoDemus** [TSAR-Industries/slashgif#18]
It would be really cool if we could automatically deploy when a PR is merged to master and all tests pass. There is support for this in [Travis](http://docs.travis-ci.com/user/deployment/), the problem is that the credentials to my server would need to be in the repo but they have [solutions for that](http://docs.travis-ci.com/user/encryption-keys/)

**Comment by Osaou** [comment]
Yeah, that solution seem spot on... I just skimmed it - but it sounded right anyway. =)
I get that this puts a bit of (boring) working on your end though, so no need to rush it, imo.

**Comment by RichoDemus** [comment]
I, after some great inspiration from oredev, was planning to use docker for this. but as long as we're hosting this on a raspberry I don't think it's feasible. for for now my plan is to create a python script which travis will run via ssh when we merge a pull request, the script will:
1. checkout the commit sent by travis
2. build it
3. stop the currently running butbut
4. backup the old binaries
5. copy the new binaries
6. start butbut again

the only concern here is security, I'll need to lock down so that the user used by travis pretty much only can do the things listed above so that the server doesn't get hacked if travis gets hacked :)

What do you guys think of the approach?
